airbyte-db/src/main/java/io/airbyte/db/bigquery/BigQueryUtils.java

```diff
@@ -0,0 +1,130 @@
+import io.airbyte.db.DataTypeUtils;
+import io.airbyte.protocol.models.JsonSchemaPrimitive;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Collections;
+import java.util.Date;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BigQueryUtils {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(BigQueryUtils.class);
+
+  public static final DateFormat BIG_QUERY_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd");
+  public static final DateFormat BIG_QUERY_DATETIME_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
+  public static final DateFormat BIG_QUERY_TIMESTAMP_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSSSS z");
+
+  public static JsonNode rowToJson(FieldValueList rowValues, FieldList fieldList) {
+    ObjectNode jsonNode = (ObjectNode) Jsons.jsonNode(Collections.emptyMap());
+    fieldList.forEach(field -> setJsonField(field, rowValues.get(field.getName()), jsonNode));
+    return jsonNode;
+  }
+
+  private static void setJsonField(Field field, FieldValue fieldValue, ObjectNode node) {
+    if (fieldValue.getAttribute().equals(Attribute.PRIMITIVE)) {
+      LegacySQLTypeName fieldType = field.getType();
+      String fieldName = field.getName();
+      if (fieldValue.isNull()) {
+        node.put(fieldName, (String) null);
+      } else
+        switch (fieldType.getStandardType()) {
+          case BOOL -> node.put(fieldName, fieldValue.getBooleanValue());
+          case INT64 -> node.put(fieldName, fieldValue.getLongValue());
+          case FLOAT64 -> node.put(fieldName, fieldValue.getDoubleValue());
+          case NUMERIC -> node.put(fieldName, fieldValue.getNumericValue());
+          case BIGNUMERIC -> node.put(fieldName, returnNullIfInvalid(fieldValue::getNumericValue));
+          case STRING -> node.put(fieldName, fieldValue.getStringValue());
+          case BYTES -> node.put(fieldName, fieldValue.getBytesValue());
+          case DATE -> node.put(fieldName, toISO8601String(getDateValue(fieldValue, BIG_QUERY_DATE_FORMAT)));
+          case DATETIME -> node.put(fieldName, toISO8601String(getDateValue(fieldValue, BIG_QUERY_DATETIME_FORMAT)));
+          case TIMESTAMP -> node.put(fieldName, toISO8601String(fieldValue.getTimestampValue() / 1000));
+          case TIME -> node.put(fieldName, fieldValue.getStringValue());
+          default -> node.put(fieldName, fieldValue.getStringValue());
+        }
+    }
+  }
+
+  public static Date getDateValue(FieldValue fieldValue, DateFormat dateFormat) {
+    Date parsedValue = null;
+    String value = fieldValue.getStringValue();
+    try {
+      parsedValue = dateFormat.parse(value);
+    } catch (ParseException e) {
+      LOGGER.error("Fail to parse date value : " + value + ". Null is returned.");
+    }
+    return parsedValue;
+  }
+
+  public static JsonSchemaPrimitive getType(StandardSQLTypeName bigQueryType) {
+    return switch (bigQueryType) {
```

Comment:
same question: anything preventing us from making this into objects and arrays?

Reply:
nope :) done